### PR TITLE
fix(CX-1959): Artist follow button

### PR DESCRIPTION
**Pull request by AlicanAkyuz**
The type of this PR is: **BugFix** and **Enhancement** 

This PR resolves [[CX-1959](https://artsyproduct.atlassian.net/browse/CX-1959)]

### Description

For some artist, our Follow button does not appear on the artist screen. The reason is that the follow button is wrapped by a conditional that checks whether artist has nationality or birthday. 

This PR: 
- display the Follow button without any condition 
- hides the followers count on artist page, if count < 2

#### IOS  
<div>
Screenshot 2021-09-03 at 14 46 09

Screenshot 2021-09-03 at 14 46 09
</div>

#### Android
<div>
Screenshot 2021-09-03 at 14 46 09

Screenshot 2021-09-03 at 14 46 09
</div>

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes
Artist page on both IOS and Android is updated by this change. 

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Artist page on both IOS and Android is updated by this change.)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ab9c8187a99e9ae60c490935750ee5269559a6c0
</p>

File: src/lib/Components/Artist/ArtistHeader.tsx [hunk 1]
```diff
@@ -119,33 +125,37 @@ export const ArtistHeader: React.FC<Props> = ({ artist, relay }) => {
...
-          </Flex>
+          )}
+          <Sans size="3t">
+            {formatLargeNumberOfItems(artist.counts?.artworks ?? 0, "work")}
+            {followersCount > 1 && (
+              <>
+                {` ${bullet} `}
+                {formatLargeNumberOfItems(followersCount, "follower")}
+              </>
+            )}
```
Comment (MounirDhahri) [inline src/lib/Components/Artist/ArtistHeader.tsx:143]: **suggestion:** You can maybe better do something like this and avoid saving the `followersCount` to the state
```suggestion
            {artist?.counts?.follows && artist.counts.follows > 1 && (
              <>
                {` ${bullet} `}
                {formatLargeNumberOfItems(followersCount, "follower")}
              </>
            )}
```

File: src/lib/Components/Artist/__tests__/ArtistHeader-tests.tsx [hunk 2]
```diff
@@ -46,6 +46,16 @@ describe("ArtistHeader", () => {
 
     expect(tree.root.findAllByType(Sans)[0].props.children).toMatch("Marcel Duchamp")
   })
+
+  it("displays follow button for artist", () => {
```
Comment (MounirDhahri) [inline src/lib/Components/Artist/__tests__/ArtistHeader-tests.tsx:50]: **issue:** To avoid the same regression in the future. Would you mind if you also add a test here to make sure that we don't show the number of followers in case there are no followers

File: src/lib/Components/Artist/ArtistArtworks/ArtistArtworks.tsx [hunk 3]
```diff
@@ -186,7 +186,12 @@ const ArtistArtworksContainer: React.FC<ArtworksGridProps & ArtistArtworksContai
     if (artworksCount === 0) {
       return (
         <Box mb="80px" pt={1}>
-          <FilteredArtworkGridZeroState id={artist.id} slug={artist.slug} trackClear={trackClear} />
+          <FilteredArtworkGridZeroState
+            id={artist.id}
+            slug={artist.slug}
+            trackClear={trackClear}
+            hideClearButton={!appliedFilters.length}
+          />
```
Comment (MounirDhahri) [inline src/lib/Components/Artist/ArtistArtworks/ArtistArtworks.tsx:194]: **thought:** I think ideally this should be part of a different PR and also linked to a Jira ticket to be able to QA in our next testing session

**Comment by MounirDhahri** [comment]
Nice one, thank you for the extra fix 🙌

File: src/lib/Components/Artist/ArtistHeader.tsx (hunk 1 above)
Comment (AlicanAkyuz) [reply src/lib/Components/Artist/ArtistHeader.tsx:143]: sure! (PS: the reason for saving it to the state is because the value is used at line 141 as well, I thought it'd be more convenient to have it in one place.)

File: src/lib/Components/Artist/ArtistArtworks/ArtistArtworks.tsx (hunk 3 above)
Comment (AlicanAkyuz) [reply src/lib/Components/Artist/ArtistArtworks/ArtistArtworks.tsx:194]: makes sense

**Comment by AlicanAkyuz** [comment]
thanks! I removed the part about hiding Clear Filters and will create another PR for it with a ticket.

File: src/lib/Components/Artist/__tests__/ArtistHeader-tests.tsx [hunk 4]
```diff
@@ -46,6 +47,30 @@ describe("ArtistHeader", () => {
...
+    const tree = renderWithWrappers(<TestRenderer />)
+
+    mockEnvironmentPayload(mockEnvironment, {
+      Artist: () => mockArtist,
+    })
+
+    expect(tree.root.findAllByType(Button)[0].props.children).toMatch("Follow")
```
Comment (pvinis) [inline src/lib/Components/Artist/__tests__/ArtistHeader-tests.tsx:58]: nice one

**Review by pvinis — APPROVED** [review]

**Review by MounirDhahri — APPROVED** [review]

**Comment by MounirDhahri** [comment]
#squashongreen
